packages/flutter/lib/src/widgets/autocomplete.dart

```diff
@@ -350,11 +350,22 @@ class _RawAutocompleteState<T extends Object> extends State<RawAutocomplete<T>>
     }
   }
 
+  int _onChangedCallId = 0;
+  int _onChangedResultId = 0;
   // Called when _textEditingController changes.
   Future<void> _onChangedField() async {
+    _onChangedCallId += 1;
+    final int callId = _onChangedCallId;
     final TextEditingValue value = _textEditingController.value;
     final Iterable<T> options = await widget.optionsBuilder(value);
+
+    // Makes sure that the previous call results do not replace to new ones
+    if (_onChangedResultId >= callId) {
+      return;
+    }
+    _onChangedResultId = callId;
```

Comment:
Sure but if i remove the _onChangedResultId the dropdown will not show the results which resolve while the user is typing in the field (it will still stop older results from overwriting the newer ones).

Without _onChangedResultId:


https://github.com/flutter/flutter/assets/95017090/4061681e-3946-43bc-98da-58110b976998


With _onChangedResultId:


https://github.com/flutter/flutter/assets/95017090/455c8ffa-d0c8-46fa-8144-5d4d239572a7


Are you sure that you want me to remove the result id?

Reply:
Ah I see. But what if you suddenly type something very different? Like type "california" and then suddenly "ctrl-a ctrl-v" and paste in "florida"? I think you might still see results for "calfornia" while "florida" is in the field.

We could put in more fanciness to handle that case, but I think maybe we should just go with my original suggestion. Maybe @victorsanni has thoughts?

Reply:
Whichever we decide, can you add another test to cover this case?

Reply:
I agree with @justinmc here. The expected behavior should be that only results pertaining to the contents of the field should be shown. If the API call is fast enough, wouldn't the results which resolve while the user is typing in the field be shown anyways?

Reply:
Ya i think you guys are right, ill get rid of the result id and update the test case